collect_app/src/test/java/org/odk/collect/android/widgets/UrlWidgetTest.java

```diff
@@ -98,26 +106,23 @@ public void clickingButton_callsCorrectMethods() {
     }
 
     @Test
-    public void onLongPressButton_doesNotCallOpenUri() {
+    public void clickingButtonForLong_callsLongClickListener() {
         UrlWidget widget = createWidget(promptWithAnswer(new StringData("blah")));
         Button urlButton = widget.findViewById(R.id.url_button);
         urlButton.performLongClick();
 
-        verify(customTabHelper, never()).bindCustomTabsService(widget.getContext(), null);
-        verify(customTabHelper, never()).openUri(widget.getContext(), Uri.parse("blah"));
+        assertThat(listener.onLongClick(urlButton), equalTo(true));
     }
 
-    @Test
-    public void textViewShouldIgnoreLongPress() {
-        UrlWidget widget = createWidget(promptWithAnswer(new StringData("blah")));
-        TextView textView = widget.findViewById(R.id.url_answer_text);
-
-        textView.onEditorAction(ACTION_UP);
-        assertThat(textView.hasFocus(), equalTo(false));
+    private TestUrlWidget createWidget(FormEntryPrompt prompt) {
+        return new TestUrlWidget(widgetTestActivity(), new QuestionDetails(prompt, "formAnalyticsID"), customTabHelper, listener);
     }
 
+    private static class TestUrlWidget extends UrlWidget {
 
-    private UrlWidget createWidget(FormEntryPrompt prompt) {
-        return new UrlWidget(widgetTestActivity(), new QuestionDetails(prompt, "formAnalyticsID"), customTabHelper);
+        public TestUrlWidget(Context context, QuestionDetails questionDetails, CustomTabHelper customTabHelper, OnLongClickListener listener) {
```

Comment:
Here's you're extending URLWidget for the test and then using that. That means your test is now of TestURLWidget not URLWidget. We shouldn't use fakes, mocks or stubs of the objects we're testing as this means we're not really testing the object itself!